lib/src/main/java/co/nimblehq/alpine/lib/mlkit/MrzProcessor.kt

```diff
@@ -0,0 +1,154 @@
+
+package co.nimblehq.alpine.lib.mlkit
+
+import android.graphics.BitmapFactory
+import android.util.Log
+import co.nimblehq.alpine.lib.mlkit.MlKitWrapperException.*
+import com.google.mlkit.common.MlKitException
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.text.*
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
+import java.io.File
+import java.util.regex.Pattern
+
+interface ResultListener {
+    fun onSuccess(mrzInfo: MrzInfo)
+
+    fun onError(e: MlKitWrapperException)
+}
+
+sealed class MlKitWrapperException : Exception() {
+    object DefaultMlKitWrapperException : MlKitWrapperException()
+    object InvalidMrzMlKitWrapperException : MlKitWrapperException()
+    object MrzNotFoundMlKitWrapperException : MlKitWrapperException()
+    object FileNotFoundMlKitWrapperException : MlKitWrapperException()
+    object TextNotFoundMlKitWrapperException : MlKitWrapperException()
+    object TextNotRecognizedMlKitWrapperException : MlKitWrapperException()
+}
+
+interface MrzProcessor {
+    fun processImageFile(filePath: String, resultListener: ResultListener)
+
+    companion object {
+        @JvmStatic
+        fun newInstance(): MrzProcessor = MrzProcessorImpl()
+    }
+}
+
+private const val TAG = "MlKitWrapper"
+private const val PASSPORT_TD_3_LINE_1_REGEX = "(P[A-Z0-9<]{1})([A-Z]{3})([A-Z0-9<]{39})"
+private const val PASSPORT_TD_3_LINE_2_REGEX =
+    "([A-Z0-9<]{9})([0-9]{1})([A-Z]{3})([0-9]{6})([0-9]{1})" +
+        "([M|F|X|<]{1})([0-9]{6})([0-9]{1})([A-Z0-9<]{14})([0-9]{1})([0-9]{1})"
+private const val IMAGE_ORIENTATION_IN_DEGREES = 90
+private const val DOCUMENT_NUMBER_MINIMUM_LENGTH = 8
+private const val DATE_OF_BIRTH_MINIMUM_LENGTH = 6
+private const val DATE_OF_EXPIRY_MINIMUM_LENGTH = 6
+
+@Suppress("NestedBlockDepth")
+private class MrzProcessorImpl : MrzProcessor {
+
+    private val textRecognizer: TextRecognizer by lazy {
+        TextRecognition.getClient(TextRecognizerOptions.DEFAULT_OPTIONS)
+    }
+    private lateinit var resultListener: ResultListener
+    private var scannedTextBuffer: String = ""
+    private var isMrzDetected: Boolean = false
+
+    @Suppress("TooGenericExceptionCaught")
+    @Throws(MlKitException::class)
+    override fun processImageFile(filePath: String, resultListener: ResultListener) {
+        isMrzDetected = false
+        this.resultListener = resultListener
+        try {
+            val file = File(filePath)
+            if (file.exists()) {
+                val bitmap = BitmapFactory.decodeFile(file.absolutePath)
+                val inputImage = InputImage.fromBitmap(bitmap, IMAGE_ORIENTATION_IN_DEGREES)
+                textRecognizer.process(inputImage)
+                    .addOnSuccessListener { results ->
+                        this.onSuccess(results)
+                    }
+                    .addOnFailureListener { e ->
+                        this.onFailure(TextNotRecognizedMlKitWrapperException)
+                    }
```

Comment:
We could simplify this block
```suggestion
                    .addOnSuccessListener { results -> onSuccess(results) }
                    .addOnFailureListener { onFailure(TextNotRecognizedMlKitWrapperException) }
```

Reply:
@ryan-conway used method reference instead. 🙏🏽 
```Kotlin
.addOnSuccessListener(::onSuccess)
```